src/todolistconverters.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module list/todolistconverters
+ */
+
+/* global document */
+
+import { generateLiInUl, injectViewList, findInRange } from './utils';
+import createElement from '@ckeditor/ckeditor5-utils/src/dom/createelement';
+
+/**
+ * @see module:engine/conversion/downcastdispatcher~DowncastDispatcher#event:insert
+ * @param {module:engine/model/model~Model} model Model instance.
+ * @returns {Function} Returns a conversion callback.
+ */
+export function modelViewInsertion( model ) {
+	return ( evt, data, conversionApi ) => {
+		const consumable = conversionApi.consumable;
+
+		if ( !consumable.test( data.item, 'insert' ) ||
+			!consumable.test( data.item, 'attribute:listType' ) ||
+			!consumable.test( data.item, 'attribute:listIndent' )
+		) {
+			return;
+		}
+
+		if ( data.item.getAttribute( 'listType' ) != 'todo' ) {
+			return;
+		}
+
+		consumable.consume( data.item, 'insert' );
+		consumable.consume( data.item, 'attribute:listType' );
+		consumable.consume( data.item, 'attribute:listIndent' );
+		consumable.consume( data.item, 'attribute:todoListChecked' );
+
+		const viewWriter = conversionApi.writer;
+		const modelItem = data.item;
+		const viewItem = generateLiInUl( modelItem, conversionApi );
+
+		addTodoElementsToListItem( modelItem, viewItem, viewWriter, model );
+		injectViewList( modelItem, viewItem, conversionApi, model );
+	};
+}
+
+/**
+ * @see module:engine/conversion/downcastdispatcher~DowncastDispatcher#event:insert
+ * @param {module:utils/eventinfo~EventInfo} evt An object containing information about the fired event.
+ * @param {Object} data Additional information about the change.
+ * @param {module:engine/conversion/downcastdispatcher~DowncastConversionApi} conversionApi Conversion interface.
+ */
+export function modelViewTextInsertion( evt, data, conversionApi ) {
+	const parent = data.range.start.parent;
+
+	if ( parent.name != 'listItem' || parent.getAttribute( 'listType' ) != 'todo' ) {
+		return;
+	}
+
+	if ( !conversionApi.consumable.consume( data.item, 'insert' ) ) {
+		return;
+	}
+
+	const viewWriter = conversionApi.writer;
+	const viewPosition = conversionApi.mapper.toViewPosition( data.range.start );
+	const viewText = viewWriter.createText( data.item.data );
+
+	viewWriter.insert( viewPosition.offset ? viewPosition : viewPosition.getShiftedBy( 1 ), viewText );
```

Comment:
A comment is needed to tell why you use `getShiftedBy( 1 )`.

Reply:
Done.